nomos-core/chain-defs/src/mantle/ledger.rs

```diff
@@ -67,52 +72,84 @@ pub struct Utxo {
     pub note: Note,
 }
 
+static NOMOS_NOTE_ID_V1: LazyLock<Fr> = LazyLock::new(|| {
+    let mut hasher = blake2::Blake2bVar::new(31).expect("blake2 initialization error");
+    hasher.update(b"NOMOS_NOTE_ID_V1");
+    let mut buff = [0u8; 31];
```

Comment:
why 31?

Reply:
Not all 32bytes vales fits in a bn254 Fr.

Reply:
yeah but a not all `Fr` fit into 31 bytes as well. Tbf the `Fr::from(BigInt)` does modular reduction so it's always possible, but we should keep an eye on that as that skews the probability so not to be used for cryptographic stuff cc @megonen @thomaslavaur

Reply:
But that would be Fr to Bytes not bytes to Fr. I get what you mean. I think we are not going Fr -> 31bytes anywhere. But lets keep the eyes open.

Reply:
Uhm, Are you sure of that? I'm missing the point. Anything 31bytes is fit into the Fr as if the remaining bytes were padded to 0 iirc. The problem would be to *dump* the Fr into 31bytes were we would have data loss.